config/v1/types_cluster_version.go

```diff
@@ -707,3 +726,89 @@ type ClusterVersionList struct {
 
 	Items []ClusterVersion `json:"items"`
 }
+
+// UpdateBundle describes updates performed using an optional update bundle.
+type UpdateBundle struct {
+	// enabled indicate if use of an update bundle is enabled. By default it is false, and the
+	// either spec.desiredUpdate.version or spec.desiredUpdate.image fields will be used to
+	// determine the required set of images images, pin and pre-load them before proceeding
+	// to apply the update. Note that in that case no bundle will be used.
+	//
+	// If explicitly set to true then the desiredUpdate.version and desiredUpdate.image fields
+	// will be ignored. Instead the cluster will loop for the update bundle and will use to
+	// determine the version, pin and pre-loaded the images and appy the update.
+	//
+	// +optional
+	Enabled bool `json:"enabled"`
+
+	// monitor indicates if the cluster should monitor device plug events to automatically
+	// detect when an update bundle is available. The default is false.
+	//
+	// +optional
+	Monitor bool `json:"monitor"`
+
+	// file indicates the location of the update bundle when the monitor fields is false. It can
+	// be an absolute file path name or an abosolute block device name.
+	//
+	// +optional
+	File string `json:"file"`
+
+	// node contains the name of the node where the update bundle is available. This is
+	// optional, and by default the cluster will look for the bundle in all the nodes.
+	//
+	// +optional
+	Node string `json:"node"`
+
+	// digest is an optional digest of the update bundle. If set then the cluster will ensure
+	// that the bundle is used only if it matches this digest. This is intended to prevent use
+	// of a the wrong bundle in the wrong cluster, or use of a perfectly fine bundle in the
+	// wrong cluster.
+	//
+	// The format is be identifier of the digest algorithm followed by a colon and the digest
+	// value.  Currently the only supported digest algorithm is sha256, so the value shold
+	// be something like sha256:1ef...03a.
```

Comment:
This will need to be validation with a kubebuilder validation and regex.
Do you expect additional types to be supported in the future?